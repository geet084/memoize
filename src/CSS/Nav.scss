.nav {
  height: 100px;
  width: 100%;
  background-color: grey;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

ul {
  list-style-type: none;
  display: flex;
}

li {
  margin: 30px;
}