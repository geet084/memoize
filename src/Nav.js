import React, { Component } from 'react';
import './CSS/Main.scss';

export default class Nav extends Component {

  render() {
    let { score, resetHandler } = this.props;
 
    return (
      <nav className="nav">
        <ul className="types">
          <li className="aTypes">Practice Array Prototypes!</li>
        </ul>
        <ul className="user">
          <li>Total correct: {score}</li>
          <li onClick={resetHandler}
            className="reset-btn">Reset your progress</li>
        </ul>
      </nav>
    );
  }
}