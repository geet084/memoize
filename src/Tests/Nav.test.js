import React from 'react';
import Nav from '../Nav.js';
import { shallow } from 'enzyme';

const mockScore = 0;
const mockResetHandler = jest.fn();

describe('Nav', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Nav
        score={mockScore}
        resetHandler={mockResetHandler} />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should register a click to reset the score', () => {
    wrapper.find('.reset-btn').simulate('click')
    expect(mockResetHandler).toBeCalled()
  })
});

