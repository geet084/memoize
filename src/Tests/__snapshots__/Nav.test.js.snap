// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with all data passed in correctly 1`] = `
<nav
  className="nav"
>
  <ul
    className="types"
  >
    <li
      className="aTypes"
    >
      Practice Array Prototypes!
    </li>
  </ul>
  <ul
    className="user"
  >
    <li>
      Total correct: 
      0
    </li>
    <li
      className="reset-btn"
      onClick={[MockFunction]}
    >
      Reset your progress
    </li>
  </ul>
</nav>
`;
